server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesResponse.java

```diff
@@ -21,20 +21,29 @@
 
 import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentFragment;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParserUtils;
 
 import java.io.IOException;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * Response for {@link FieldCapabilitiesRequest} requests.
  */
 public class FieldCapabilitiesResponse extends ActionResponse implements ToXContentFragment {
```

Comment:
Would it be possible to make this a `ToXContentObject`? then we could also adapt the corresponding REST action and use `RestToXContentListener` in there.

Reply:
yes sure we usually make this change as part these PRs mostly because in many cases we also go from `ToXContentFragment` to `ToXContentObject`. I am fine with a follow-up PR.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/30182.